src/basilisp/core.lpy

```diff
@@ -7215,3 +7186,104 @@
      (.put-nowait tap-queue {:topic topic :val val})
      true
      (catch queue/Full _ false))))
+
+(defprotocol SeqReduce
+  "Protocol impl of reduce that's faster than first/next iteration. Called by reduce."
+  (seq-reduce-impl [s f init]))
+
+(extend-protocol SeqReduce
+  nil
+  (seq-reduce-impl [s f init] init)
+
+  basilisp.lang.interfaces/ILispObject
```

Comment:
I'm unsure of what the intent is behind this extension. `ILispObject` is basically just an interface about generating textual representations of objects (in the spirit of Python `repr`). Do you mean `python/object` (the Python top type and equivalent to `java.lang.Object`)?

Reply:
Oo, yes I did. I forgot about Python's Object. I'll update.